tests/test_kernels.py

```diff
@@ -10,22 +10,36 @@ def grid_type_field_and_extra_kwargs(request):
 
     ny, nx = (128, 256)
     data = np.random.rand(ny, nx)
+    mask_data = np.ones_like(data)
+    mask_data[: (ny // 2), : (nx // 2)] = 0
 
     extra_kwargs = {}
     if grid_type == GridType.CARTESIAN_WITH_LAND:
-        mask_data = np.ones_like(data)
-        mask_data[: (ny // 2), : (nx // 2)] = 0
         extra_kwargs["wet_mask"] = mask_data
     if grid_type == GridType.IRREGULAR_CARTESIAN_WITH_LAND:
-        mask_data = np.ones_like(data)
-        mask_data[: (ny // 2), : (nx // 2)] = 0
         extra_kwargs["wet_mask"] = mask_data
         grid_data = np.ones_like(data)
         extra_kwargs["dxw"] = grid_data
         extra_kwargs["dyw"] = grid_data
         extra_kwargs["dxs"] = grid_data
         extra_kwargs["dys"] = grid_data
         extra_kwargs["area"] = grid_data
+    if (grid_type == GridType.MOM5U) or (grid_type == GridType.MOM5T):
+        dxu, dyu = np.meshgrid(np.random.rand(nx), np.random.rand(ny))
+        dxt, dyt = dxu, dyu
+        extra_kwargs["wet"] = mask_data
+        grid_data = np.ones_like(data)
+        # dxu, dyu = grid_data, grid_data
+        # dxt, dyt = grid_data, grid_data
+        extra_kwargs["wet"] = np.ones_like(data)
+        extra_kwargs["dxu"] = dxu
+        extra_kwargs["dyu"] = dyu
+        extra_kwargs["dxt"] = dxt
+        extra_kwargs["dyt"] = dyt
```

Comment:
It's great that you are creating grid data that is actually **irregular**. I will adopt your test for my `IRREGULAR_CARTESIAN_WITH_LAND` a few lines above. :smile: With irregular grid data, the `test_conservation` test has to be changed to an area-integrated integral, something like:
```
res = (diffused * extra_kwargs["area"]).sum()
```

 This may be part of the reason why the test_conservation test fails for your MOM5 Laplacians.

Reply:
Ah yes of course, thanks I'll give it a try.